libp2p/protocols/connectivity/autonatv2/client.nim

```diff
@@ -0,0 +1,117 @@
+# Nim-LibP2P
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+
+import results
+import chronos, chronicles
+import
+  ../../protocol,
+  ../../../switch,
+  ../../../multiaddress,
+  ../../../multicodec,
+  ../../../peerid,
+  ../../../protobuf/minprotobuf,
+  ./types
+
+logScope:
+  topics = "libp2p autonat v2 client"
+
+const
+  MaxAcceptedDialDataRequest* = 100 * 1024 # 100 KB
+  MaxDialDataResponsePayload* = 1024
+
+type AutonatV2Client* = ref object of RootObj
+  rng: ref HmacDrbgContext
+  nonces: Table[PeerId, uint64]
+
+proc new*(T: typedesc[AutonatV2Client], rng: ref HmacDrbgContext, switch: Switch): T =
+  let client = T(rng: rng, nonces: initTable[PeerId, uint64]())
+  client
+
+proc handleDialDataRequest*(
```

Comment:
`handleDialDataRequest` anyone calling this method has to do two things to handle error:
 - check if `Opt` values is set
 - `try/except` to handle errors 
 
 how come `handleDialDataRequest` can return `Opt.none(...)` and that is not error case?